src/bases.d.ts

```diff
@@ -58,7 +58,7 @@ export interface Evented extends Destroyable {
 	 * @param listener A listener or array of listeners that accept error events
 	 * @returns A handle which can be used to remove the listeners
 	 */
-	on(type: 'error', listener: EventedListenerOrArray<this, EventErrorObject<this>>): Handle;
+	on<T>(type: 'error', listener: EventedListenerOrArray<T, EventErrorObject<T>>): Handle;
```

Comment:
or `T extends Evented`?  I guess I am just trying to make it as narrow as possible so there is less need for downstream developers to be explicit.  The rule would be that events should be at least as narrow as the class they are declared in.